### PR TITLE
chore(iast): re.subn and re.split aspects

### PR DESCRIPTION
**Pull request by gnufede**
Code security: Aspects for `re.subn` and `re.split`

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/appsec/app.py [hunk 1]
```diff
@@ -242,6 +242,90 @@ def iast_ast_patching_non_re():
...
+
+        if is_pyobject_tainted(changed):
+            resp = Response("OK")
+    except Exception as e:
+        print(e)
+    return resp
+
+
+@app.route("/iast-ast-patching-non-re-subn", methods=["GET"])
+def iast_ast_patching_non_re_subn():
```
Comment (avara1986) [inline tests/appsec/app.py:266]: ast_ast_patching_re_subn and iast_ast_patching_non_re_subn are the same function, are they?

why not?
```python
@app.route("/iast-ast-patching-non-re-subn", methods=["GET"])
@app.route("/iast-ast-patching-re-subn", methods=["GET"])
def iast_ast_patching_re_subn():
   ...
```

File: tests/appsec/app.py [hunk 2]
```diff
@@ -242,6 +242,90 @@ def iast_ast_patching_non_re():
...
+
+        if all(map(is_pyobject_tainted, result)):
+            resp = Response("OK")
+    except Exception as e:
+        print(e)
+    return resp
+
+
+@app.route("/iast-ast-patching-non-re-split", methods=["GET"])
+def iast_ast_patching_non_re_split():
```
Comment (avara1986) [inline tests/appsec/app.py:308]: ditto

File: tests/appsec/app.py [hunk 3]
```diff
@@ -220,8 +220,8 @@ def iast_ast_patching_re():
     return resp
 
 
-@app.route("/iast-ast-patching-non-re", methods=["GET"])
-def iast_ast_patching_non_re():
+@app.route("/iast-ast-patching-non-re-sub", methods=["GET"])
+def iast_ast_patching_non_re_sub():
```
Comment (avara1986) [inline tests/appsec/app.py:224]: `iast_ast_patching_re_sub` and `iast_ast_patching_non_re_sub` are the same function, are they?

why not?

@app.route("/iast-ast-patching-re-sub", methods=["GET"])
@app.route("/iast-ast-patching-non-re-sub", methods=["GET"])
def iast_ast_patching_re_sub():
   ...

Comment (gnufede) [reply tests/appsec/app.py:224]: They're not the same, non-re-sub does this extra line at the top of the function:
```
    import iast.fixtures.non_re_module as re
```

Comment (gnufede) [reply tests/appsec/app.py:224]: We could do that under an if depending on some request.args though

**Review by avara1986 — APPROVED** [review]

File: tests/appsec/app.py (hunk 1 above)
Comment (gnufede) [reply tests/appsec/app.py:266]: ditto

File: tests/appsec/app.py (hunk 2 above)
Comment (gnufede) [reply tests/appsec/app.py:308]: ditto

File: tests/appsec/app.py (hunk 3 above)
Comment (gnufede) [reply tests/appsec/app.py:224]: Tested with the if, it doesn't work as expected

**Pull request merged**